Kernel/Memory/MemoryManager.cpp

```diff
@@ -278,152 +280,370 @@ UNMAP_AFTER_INIT void MemoryManager::parse_memory_map()
 #endif
         global_data.used_memory_ranges.append(UsedMemoryRange { UsedMemoryRangeType::Kernel, PhysicalAddress(virtual_to_low_physical((FlatPtr)start_of_kernel_image)), PhysicalAddress(page_round_up(virtual_to_low_physical((FlatPtr)end_of_kernel_image)).release_value_but_fixme_should_propagate_errors()) });
 
-        if (multiboot_flags & 0x4) {
-            auto* bootmods_start = multiboot_copy_boot_modules_array;
-            auto* bootmods_end = bootmods_start + multiboot_copy_boot_modules_count;
+#if ARCH(RISCV64)
+        parse_memory_map_fdt(global_data, s_boot_info.fdt_phys_addr);
```

Comment:
We probably want to use this on AArch64 as well asap... can there be a FIXME here?